app/controllers/show_projects_controller.rb

```diff
@@ -0,0 +1,6 @@
+class ShowProjectsController < ApplicationController
```

Comment:
Hm, I think I don't really get the point here 🙈 
There is already a public projects page - which in my opinion also makes sense, students applying need to have an overview of projects available.

So what is the difference in information this new page now offers? If it's something only a specific group of users should be allowed to see, then I'd vote for a nested controller in the respective users' role namespace. If this is the general *"project page"* than I'd say it's fine to be non-restricted and therefore not nested into any role namespace.